.github/workflows/code_quality.yml

```diff
@@ -0,0 +1,52 @@
+name: 🚧 Lint and run test suite
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
The lack of explicitly listed activity types caught my attention. But from the [pull request trigger docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request), the default triggers are exactly the desired behavior.

> if no activity types are specified, the workflow runs when a pull request is opened or reopened or when the head branch of the pull request is updated.

Reply:
Yup exactly. I also thought about adding `[opened, synchronized, reopened]` but figured we can keep it as is until we need to customize further